extensions/default/src/utils/reuseCachedLayouts.ts

```diff
@@ -58,8 +58,10 @@ const reuseCachedLayout = (state, hangingProtocolService: HangingProtocolService
       if (!displaySetUID) {
         continue;
       }
-      if (viewportId === activeViewportId && i === 0) {
-        setDisplaySetSelector(`${activeStudyUID}:activeDisplaySet:0`, displaySetUID);
+      if (viewportId === activeViewportId) {
+        // For the active viewport, overload the `matchedDisplaySetsIndex` position in the cache key
```

Comment:
@wayfarer3130, please have a look at this and let me know what you think. This is one way to cache the various overlay display sets for the active viewport. The other thing I can think of is to add overlays to hanging protocols which does not make much sense because in essence multiple display sets can already be defined for a viewport in a hanging protocol. The overlays appear to be a "view only"/"in app only" feature. One other thing could be to expand the cache key here to include the overlay index separately from the `matchedDisplaySetsIndex`.